### PR TITLE
chore: Simpler Future<> without locking

### PR DESCRIPTION
**Pull request by dranikpg**
Introduce simple future to replace boost code

replacing aligned_storage with optional, intrusive_ptr with shared_ptr and a mutex with event-count makes the code ten times smaller 🙂

### CODE REVIEW

**Comment by dranikpg** [comment]
Actually this is unsafe, need to fix this 😅

File: util/fibers/future.h [hunk 1]
```diff
@@ -4,339 +4,39 @@
...
-    base_type::future_->set_value(value);
-  }
-
-  void swap(Promise& other) noexcept {
-    base_type::swap(other);
-  }
-
-  using base_type::get_future;
+  struct Block {
+    T value{};  // replace with aligned_storage or optional if T is not default-constructible
```
Comment (romange) [inline util/fibers/future.h:35]: consider doing this.

File: util/fibers/future.h [hunk 2]
```diff
@@ -4,339 +4,39 @@
...
+    std::atomic_bool has_value = false;
+    util::fb2::EventCount waker{};
+  };
+  std::shared_ptr<Block> block;
```
Comment (romange) [inline util/fibers/future.h:39]: if you extend to  `std::variant<std::shared_ptr<Block>, T>` you could provide support for immediate future built-in.

**Review by romange — APPROVED** [review]

File: util/fibers/future.h (hunk 2 above)
Comment (dranikpg) [reply util/fibers/future.h:39]: I already have StringValue in the main repo

**Pull request merged**